### PR TITLE
chore: Explicit plan for no grouping columns when none are provided.

### PR DESCRIPTION
**Pull request by benclive**
**What this PR does / why we need it**:
When planning a query like `sum(count_over_time({container="loki"}[5m]))` the range aggregation only needs timestamps as there is no By or Without clause.

The previous behaviour would default to reading all columns (e.g. `without ()`) rather than reading timestamps only. This fixes that case.

In my test query, this removed any reading of secondary bytes which was 90% of the bytes loaded.

---

> [!NOTE]
> **Medium Risk**
> Changes the default grouping semantics for metric aggregations, which affects planning/projection pushdown and could alter which columns are scanned for some queries. Main risk is behavioral differences in edge cases around `by/without` handling, though coverage is added via updated planner tests.
> 
> **Overview**
> Fixes metric planning so *absence of a `by`/`without` clause* is represented as explicit `group_by=()` (via `NoGrouping`) instead of behaving like `without ()`, enabling projection pushdown to scan only required columns (notably timestamp-only for ungrouped range aggregations).
> 
> Updates logical-plan string expectations and adds a test for empty grouping, plus adjusts projection pushdown comments/behavior to treat `RangeAggregation` projections as only needed for `by` groupings (skipping `without`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 433a7d1902403304af72c28a06357e7d6f12bb44. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: pkg/engine/internal/planner/logical/planner_test.go [hunk 1]
```diff
@@ -295,7 +331,42 @@ RETURN %12
...
+		require.NoError(t, err)
+		t.Logf("\n%s\n", logicalPlan.String())
+
+		expected := `%1 = EQ label.cluster "prod"
+%2 = MAKETABLE [selector=%1, predicates=[], shard=0_of_1]
+%3 = GTE builtin.timestamp 1970-01-01T00:55:00Z
+%4 = SELECT %2 [predicate=%3]
+%5 = LT builtin.timestamp 1970-01-01T02:00:00Z
+%6 = SELECT %4 [predicate=%5]
+%7 = RANGE_AGGREGATION %6 [group_by=(), operation=count, start_ts=1970-01-01T01:00:00Z, end_ts=1970-01-01T02:00:00Z, step=0s, range=5m0s]
```
Comment (ashwanthgoli) [inline pkg/engine/internal/planner/logical/planner_test.go:369]: this should group by `level` right? there is a missing change in groupBy pushdown i thiunk
